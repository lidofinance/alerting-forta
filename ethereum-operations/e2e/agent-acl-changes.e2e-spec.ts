import { configureContainer, Finding } from "forta-agent";
import { AwilixContainer, asFunction } from "awilix";
import {
  provideAgentPath,
  provideRunBlock,
  provideRunTransaction,
} from "./utils";

const TEST_TIMEOUT = 60_000; // ms

describe("agent-acl-changes e2e tests", () => {
  let runBlock: (blockHashOrNumber: string | number) => Promise<Finding[]>;
  let runTransaction: (txHash: string) => Promise<Finding[]>;
  let logSpy: jest.SpyInstance;

  beforeAll(() => {
    logSpy = jest.spyOn(console, "log");
    logSpy.mockImplementation(() => {});
  });

  beforeEach(async () => {
    const container = configureContainer() as AwilixContainer;
    container.register({
      agentPath: asFunction(
        provideAgentPath("subagents/acl-changes/agent-acl-changes")
      ),
      runTransaction: asFunction(provideRunTransaction),
      runBlock: asFunction(provideRunBlock),
    });

    // https://docs.forta.network/en/latest/cli/#invoke-commands-programmatically
    runTransaction = container.resolve("runTransaction");
    runBlock = container.resolve("runBlock");
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it(
    "should process tx with permission change",
    async () => {
      const findings = await runTransaction(
        "0x46d937a9bb533feaf3b7936d230822eecc65d7ff4f6e38a4e17d3ca59cdf0799"
      );
      expect(findings.at(0)).toMatchSnapshot();
    },
    TEST_TIMEOUT
  );

  it(
    "should process tx with permission manager changed",
    async () => {
      const findings = await runTransaction(
        "0x11a48020ae69cf08bd063f1fbc8ecf65bd057015aaa991bf507dbc598aadb68e"
      );
      expect(findings.at(2)).toMatchSnapshot();
    },
    TEST_TIMEOUT
  );
});
