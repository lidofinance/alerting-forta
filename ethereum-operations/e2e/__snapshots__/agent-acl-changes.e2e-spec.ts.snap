// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agent-acl-changes e2e tests should process tx with permission change 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "ARAGON-ACL-PERMISSION-CHANGED",
  "description": "Role 0x5de467a460382d13defdc02aacddc9c7d6605d6d4e0b8bd2f70732cae8ea17bc (CREATE PAYMENTS ROLE) on the app [0xB9E5CBB9CA5b0d659238807E84D0176930753d86](https://etherscan.io/address/0xB9E5CBB9CA5b0d659238807E84D0176930753d86) (Aragon Finance) was revoked from and granted with params to 0xFE5986E06210aC1eCC1aDCafc0cc7f8D63B3F977 (EVMScriptExecutor)",
  "labels": Array [],
  "metadata": Object {},
  "name": "ðŸš¨ Aragon ACL: Permission revoked and granted with params",
  "protocol": "ethereum",
  "severity": 3,
  "type": 4,
}
`;

exports[`agent-acl-changes e2e tests should process tx with permission manager changed 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "ARAGON-ACL-PERMISSION-MANAGER-CHANGED",
  "description": "Permission manager for the role 0x9d68ad53a92b6f44b2e8fb18d211bf8ccb1114f6fafd56aa364515dfdf23c44f (SET EL REWARDS VAULT ROLE) on the app [0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84](https://etherscan.io/address/0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84) (stETH token) was set to [0x2e59A20f205bB85a89C53f1936454680651E618e](https://etherscan.io/address/0x2e59A20f205bB85a89C53f1936454680651E618e) (Aragon Voting)",
  "labels": Array [],
  "metadata": Object {},
  "name": "ðŸš¨ Aragon ACL: Permission manager changed",
  "protocol": "ethereum",
  "severity": 5,
  "type": 4,
}
`;
